### ISSUE TITLE
Espanso not starting Windows 10

### ISSUE THREAD

**Issue by Goobergoobs** [federico-terzi/espanso#667]
Hello. I recently added a new trigger/replace and since this Espanso will not start anymore. I uninstalled, restart, then reinstalled but it still will not work. When attempting to run Espanso, nothing happens. No windows, no pop-ups, nothing. Its as if it was never run in the first place. I have also checked my Windows Security settings to see if maybe it was blocking the application but nothing shows up in the history. Espasno is still enabled in my startup applications, but it still does not load even after restarting. Thanks in advance for any help on this matter!

**Comment by databurstrp** [comment]
+1 this is also happening to me, as soon as I added a new trigger. I removed the new addition to the .yml file and restarted but it still won't load. The new trigger was an identical format to the dozen others I've already put in that have been working fine.

**Comment by trebsoj** [comment]
+1

**Comment by federico-terzi** [comment]
Hey guys,

I'm sorry you experienced this problem! Could you please post the output of `espanso log` here? Also, what Windows version are you using?

Cheers :)

**Comment by Goobergoobs** [comment]
Unfortunately I do no see a log file anywhere. If you could let me know where it is then I can upload it for you. As far as Windows version. I am using Windows 10 Pro with the 20H2 Update.

**Comment by federico-terzi** [comment]
@Goobergoobs Thank you

What I meant was opening a terminal, type `espanso log` and then post the output here :)

**Comment by JeppeKlitgaard** [comment]
I think this is likely just some mangled configuration file.

What I do to find where the error in my YAML configurations is, is to simply run `espanso` in a terminal. This will helpfully point you to the line and column where the YAML parsing fails.

Perhaps `espanso` should put the error message in a pop-up when it fails on startup. For non-terminal users, it currently fails silently. Alternatively/in addition the 'trick' of running `espanso` to find the error could be added as a line or two in the documentation.

**Comment by TSSlade** [comment]
@JeppeKlitgaard - it's not just the configuration file. As @databurstrp indicated, restoring the 'offending' ('mangled', as you put it) config file does not get back to full function, even with a restart.

I can also affirm @Goobergoobs' experience that attempting to trigger espanso from the CLI also fails silently. See #591 for similar behavior and the insufficiency of invoking espanso from the CLI as a diagnostic method.

**Comment by JeppeKlitgaard** [comment]
@TSSlade 

Fair enough - I will leave my initial comment, as I do think this might be the case for some users who find their way to this thread. As far as I can tell, there aren't any symlinks in play here, so I am not entirely sure you can diagnose these two issues as having the same root course.

**Comment by TSSlade** [comment]
I agree that checking the config file is a great place to start! And also that the underlying causes are quite possibly very different.
Just wanted to make clear that the failure mode described here is not unique to mangled config files.

**Comment by federico-terzi** [comment]
@JeppeKlitgaard @TSSlade Thank you for the feedback guys! Those points are definitely valid, as the current behavior (silently failing when the YAML is invalid) is far from being user-friendly.
I'm really looking forward to improving this aspect in #594, though it might take a few more minor releases :)

**Comment by federico-terzi** [comment]
Hey guys,

The new alpha v2 release automatically shows a "troubleshooting" window if a configuration error is detected. That way, it should be much easier for the user to debug this kind of problems!

![image](https://user-images.githubusercontent.com/20475953/136710444-9f037bcf-3b72-4a00-8255-ce91ce71a683.png)

Again, thank you for the valuable feedback!

I think we can safely close this issue now, but feel free to reopen if you encounter any other problems or have some feedback! :)

Cheers :)
